### PR TITLE
make containerized make targets support alternative container runtimes

### PR DESCRIPTION
**Pull request by tal-hason**
as of #905 

i have update the _make_ file to work with an environment variable CONTAINER_RUNTIME, replaced it with the "docker" in the files

 at the beginning of the _make_ file I have added a bash script that checks if podman or docker exists and stores it in the CONTAINER_RUNTIME env. var

### CODE REVIEW

**Comment by tal-hason** [comment]
i can replace the order the docker will be the first and podman the second

or 

make just as the env var with remarks

your call

**Comment by krancour** [comment]
The env var... the thing is that I anticipate there will be plenty of people who prefer one or the other but happen to have both for... reasons.

**Comment by tal-hason** [comment]
No prob, I will patch that later

File: Makefile [hunk 1]
```diff
@@ -1,12 +1,16 @@
...
-ARGO_CD_CHART_VERSION := 5.46.6
+ARGO_CD_CHART_VERSION := 5.46.3
 BUF_LINT_ERROR_FORMAT ?= text
 GO_LINT_ERROR_FORMAT ?= colored-line-number
 CERT_MANAGER_CHART_VERSION := 1.11.5
 
 VERSION_PACKAGE := github.com/akuity/kargo/internal/version
 
+# default docker , change to podman if you using podman
+CONTAINER_RUNTIME = docker
```
Comment (krancour) [inline Makefile:11]: Did you test this? This line sets it to `docker` unconditionally. i.e. Whatever is in the env var gets overwritten here.

https://www.gnu.org/software/make/manual/html_node/Setting.html

```suggestion
CONTAINER_RUNTIME ?= docker
```

File: Makefile [hunk 2]
```diff
@@ -1,12 +1,16 @@
 SHELL ?= /bin/bash
 
-ARGO_CD_CHART_VERSION := 5.46.6
+ARGO_CD_CHART_VERSION := 5.46.3
```
Comment (krancour) [inline Makefile:3]: Why did you change this?

File: Makefile [hunk 3]
```diff
@@ -1,12 +1,16 @@
...
 BUF_LINT_ERROR_FORMAT ?= text
 GO_LINT_ERROR_FORMAT ?= colored-line-number
 CERT_MANAGER_CHART_VERSION := 1.11.5
 
 VERSION_PACKAGE := github.com/akuity/kargo/internal/version
 
+# default docker , change to podman if you using podman
```
Comment (krancour) [inline Makefile:10]:
```suggestion
# Default to docker, but support alternative container runtimes that are CLI-compatible with Docker
```

File: Makefile (hunk 2 above)
Comment (tal-hason) [reply Makefile:3]: > Why did you change this?

trying to bootstrap the kind and the argocd repo server is crash-looping something with the Redis , so I tested it with a older version and for got to change it, sorry

File: docs/docs/40-contributor-guide/10-hacking-on-kargo.md [hunk 4]
```diff
@@ -130,6 +130,7 @@ The remainder of this section covers the tools we recommend for enabling this.
 You will need:
 
 * [Docker](https://docs.docker.com/engine/install/)
+  > you can use podman by changing the *CONTAINER_RUNTIME_ in the make file from docker to podman, [How to config podman socket](https://docs.podman.io/en/latest/markdown/podman-system-service.1.html#examples)
```
[the remark this answers is not in the document]
Comment (tal-hason) [reply docs/docs/40-contributor-guide/10-hacking-on-kargo.md:133]: where does it need to be?

[the remark this answers is not in the document]
Comment (krancour) [reply docs/docs/40-contributor-guide/10-hacking-on-kargo.md:133]: This section is about running _everything_ using Tilt. idk if Tilt even works with podman. 🤷 

There's a section much higher up, around line 12, about running tests and other contributor-facing tasks in a container.

[the remark this answers is not in the document]
Comment (krancour) [reply docs/docs/40-contributor-guide/10-hacking-on-kargo.md:133]: Here's how it can work with Tilt.

https://docs.tilt.dev/custom_build.html#buildah-or-any-image-builder-independent-of-docker

If you want to do that also, separate issue/PR please.

File: docs/docs/40-contributor-guide/10-hacking-on-kargo.md [hunk 5]
```diff
@@ -18,6 +18,13 @@ execute tests within a container that is maximally similar to those used in CI.
 
 To take advantage of this, you only need to have
 [Docker](https://docs.docker.com/engine/install/) and `make` installed.
+> you can use podman by creating an environment variable
+
+```bash
+export CONTAINER_RUNTIME=podman
+```
+
+[How to config podman socket](https://docs.podman.ioen/latest/markdown/podman-system-service.1.html#examples)
```
Comment (krancour) [inline docs/docs/40-contributor-guide/10-hacking-on-kargo.md:27]: Do we need to be podman-specific here?

Comment (krancour) [reply docs/docs/40-contributor-guide/10-hacking-on-kargo.md:27]: > Do we need to be podman-specific here?

I realized my meaning might have been ambiguous.

The comment _wasn't_ about the podman socket setup.

It was about the general idea of this PR adding _prodman_ support.

I'm fine with the notion of supporting alternative container runtimes as long as they are "Docker compatible," but I'm less ok with presenting podman as a specially preferred/specially supported alternative container runtime... if that makes any sense.

I think the easiest thing to do here might be to just not mention this in the docs. The changes to the `Makefile` are good and I think anyone who really cares about using an alternative container runtime will find the way of doing it easily enough.

**Comment by krancour** [comment]
@tal-hason idk what happened with this PR, but it seems to include changes that are already in `main`. I'm not sure how that would have happened. Do you want to force push just the relevant commit or two to here? Alternatively, you could close this PR and open a new one having just the relevant changes.

Sorry... really no idea how it ended up in this odd state.

**Comment by tal-hason** [comment]
> @tal-hason idk what happened with this PR, but it seems to include changes that are already in `main`. I'm not sure how that would have happened. Do you want to force push just the relevant commit or two to here? Alternatively, you could close this PR and open a new one having just the relevant changes.
> 
> Sorry... really no idea how it ended up in this odd state.

weird , OK I will reopen the PR

**Comment by tal-hason** [comment]
> @tal-hason idk what happened with this PR, but it seems to include changes that are already in `main`. I'm not sure how that would have happened. Do you want to force push just the relevant commit or two to here? Alternatively, you could close this PR and open a new one having just the relevant changes.
> 
> Sorry... really no idea how it ended up in this odd state.

i can make a new branch from a fresh and just add the changes, maybe later on

**Comment by tal-hason** [comment]
Opened #950 PR with a fresh branch
